### PR TITLE
Support having the snapcraft.yaml in a subdir.

### PR DESCRIPTION
**Pull request by kyrofa**
Currently when one puts the `snapcraft.yaml` in a subdirectory of the project (using `source: ..` etc.) snapcraft dies in a sea of recursion. This PR fixes LP: [#1600428](https://bugs.launchpad.net/snapcraft/+bug/1600428) by skipping over the directory containing the snapcraft.yaml if it's a subdirectory of the project.

### CODE REVIEW

**Comment by evandandrea** [comment]
Thanks for picking this up! Sorry I couldn't find the time for those integration tests.

File: integration_tests/test_yaml_in_subdir.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+        expected_file = os.path.join(subdir, 'stage', 'file')
+        self.assertThat(expected_file, FileExists())
+        with open(expected_file) as f:
+            self.assertThat(f.read(), Equals("I'm a file\n"))
+
+        expected_file = os.path.join(subdir, 'stage', 'subdirfile')
+        self.assertThat(expected_file, FileExists())
+        with open(expected_file) as f:
+            self.assertThat(f.read(), Equals("I'm in the subdir\n"))
```
Comment (elopio) [inline integration_tests/test_yaml_in_subdir.py]: :tada:

**Comment by elopio** [comment]
:+1:

**Comment by elopio** [comment]
retest this please

**Comment by kyrofa** [comment]
retest this please.

**Comment by kyrofa** [comment]
retest this please

**Pull request merged**